tools/catsitemap.py

```diff
@@ -0,0 +1,119 @@
+"""Script for automatic generation/download of sitemaps for pyansys projects.
+
+Intended for the pyansys project and to be used with the update-gh-pages
+workflow.
+
+"""
+
+from pathlib import Path
+from xml.dom import minidom
+import xml.etree.ElementTree as ET
+
+from links import LINKS
+import requests
+
+
+def download_file(url: str, dest_path: Path) -> None:
+    """Given a sitemap url, this function downloads the file into (dest_path).
+
+    Parameters
+    ----------
+    url : str
+        The url of the sitemap file to be downloaded
+    dest_path : Path
+        The destination path to save the downloaded file
+
+    Raises
+    ------
+    requests.exceptions.Timeout
+        Raises this exception when accessing a link takes too long
+    """
+    # Send the request
+    try:
+        response = requests.get(url, stream=True, timeout=30)
+    except requests.exceptions.Timeout as e:
+        print(f"Timed out while trying to get download the sitemap at this url: {url}")
+        raise e
+
+    # Write the file content to the specified location
+    with open(dest_path, mode="wb") as file:
+        for chunk in response.iter_content(chunk_size=8192):
```

Comment:
If the chunk_size is a constant, it would be better to pass it as a parameter. We can also set its default value at the top of the file in the form of a constant. Use upper case for its name if you decide to proceed with that.

Reply:
I removed the need for "chunks" and the for loop altogether. Chunk size variable wasn't needed anymore.